### PR TITLE
#1444 FormBuilder. File input

### PR DESCRIPTION
**Pull request by BALEHOK**
Related to #1444

### CODE REVIEW

File: src/components/form/widgets/SelectWidget.tsx [hunk 1]
```diff
@@ -21,6 +21,7 @@ import Select from "react-select";
 import { getErrorMessage } from "@/errors";
 
 export type Option<TValue = unknown> = {
+  id?: string;
```
Comment (twschiller) [inline src/components/form/widgets/SelectWidget.tsx:24]: What is the id on the option for? Is this a typo? (And meant to be on OwnProps instead?

- [ ] Fix is on Option type, or document why it's there

**Review by twschiller — CHANGES_REQUESTED** [review]
See comment about the id field on the Option type

File: src/components/formBuilder/FieldEditor.tsx [hunk 2]
```diff
@@ -143,17 +144,21 @@ const FieldEditor: React.FC<{
       const draftPropertySchema = draft.schema.properties[
         propertyName
       ] as Schema;
-      if (propertySchema.type !== propertyType) {
```
Comment (twschiller) [inline src/components/formBuilder/FieldEditor.tsx:146]: This method is long enough now that you should consider extracting it to a helper method. That would also make it easier to test all the conditions/corner cases in unit tests

File: src/components/form/widgets/SelectWidget.tsx (hunk 1 above)
Comment (BALEHOK) [reply src/components/form/widgets/SelectWidget.tsx]: It was a typo. Should be on Widget props but in fact `CustomFieldWidget` extends `FieldProps`, which extends `FormControlProps`, which already has `id`.

**Review by twschiller — APPROVED** [review]
